pages/resources/components.mdx

```diff
@@ -0,0 +1,1364 @@
+| type      | integer                                        | `2` for a button                                                                  |
+| id?       | integer                                        | Optional identifier for component                                                 |
+| style     | integer                                        | A [button style](#button-styles)                                                  |
+| label?    | string                                         | Text that appears on the button; max 80 characters                                |
+| emoji?    | partial [emoji](/resources/emoji#emoji-object) | `name`, `id`, and `animated`                                                      |
+| custom_id | string                                         | Developer-defined identifier for the button; max 100 characters                   |
+| sku_id?   | snowflake                                      | Identifier for a purchasable SKU, only available when using premium-style buttons |
+| url?      | string                                         | URL for link-style buttons                                                        |
+| disabled? | boolean                                        | Whether the button is disabled (defaults to `false`)                              |
+
+Buttons come in various styles to convey different types of actions. These styles also define what fields are valid for a button.
+
+- Non-link and non-premium buttons **must** have a `custom_id`, and cannot have a `url` or a `sku_id`.
+- Link buttons **must** have a `url`, and cannot have a `custom_id`
+- Link buttons do not send an [interaction](/interactions/receiving-and-responding#interaction-object) to your app when clicked
+- Premium buttons **must** contain a `sku_id`, and cannot have a `custom_id`, `label`, `url`, or `emoji`.
+- Premium buttons do not send an [interaction](/interactions/receiving-and-responding#interaction-object) to your app when clicked
+
+###### Button Styles
+
+| Name      | Value | Action                                                         | Required Field |
+| --------- | ----- | -------------------------------------------------------------- | -------------- |
+| Primary   | 1     | The most important or recommended action in a group of options | `custom_id`    |
+| Secondary | 2     | Alternative or supporting actions                              | `custom_id`    |
+| Success   | 3     | Positive confirmation or completion actions                    | `custom_id`    |
+| Danger    | 4     | An action with irreversible consequences                       | `custom_id`    |
+| Link      | 5     | Navigates to a URL                                             | `url`          |
+| Premium   | 6     | Purchase                                                       | `sku_id`       |
```

Comment:
should be Value | Name | Description | Required Field and names should be capital